### DISCUSSION TITLE
The Compiled exe gets undeletable

### DISCUSSION THREAD

**Discussion by RandomTimeLP** [denoland/deno#20551] [Q&A]
Trying to implement Webviewer i discovered that i can not implement Webviewer as a function. so i've tried to implemet it this way (it does work, opens the window properly):
```
if(!existsSync("./ui.ts")) {
  console.log("creating ui.ts")
await Deno.writeTextFile("./ui.ts", `import { Webview } from "https://deno.land/x/webview@0.7.6/mod.ts";
const webview = new Webview();
webview.navigate(\`http://localhost:1992/\`);
webview.run();`)
}
if(existsSync("./ui.ts")) {
  const cmd = new Deno.Command(Deno.execPath(), {
    args: ["run", "-A", "--unstable", `./ui.ts`],
  });
  cmd.output();
} 
```
However this method bricks the program (if executed compiled), rendering it undeletable and unusable until the computer gets restarted.

https://youtu.be/l4Y3n0OBPug => the video displaying the issue

**Comment by lucacasonato** [comment]
You should add a signal listener to listen for shutdown of the main process, and if it shuts down, run `cmd.kill("SIGTERM")`. Sub processes are not automatically killed when the parent process dies.
